cpp/src/glm/ridge.cuh

```diff
@@ -155,12 +153,18 @@ void ridgeFit(const raft::handle_t &handle, math_t *input, int n_rows,
   ASSERT(n_rows > 1, "ridgeFit: number of rows cannot be less than two");
 
   math_t *mu_input, *norm2_input, *mu_labels;
+  rmm::device_uvector<math_t> mu_input_vector(0, stream);
+  rmm::device_uvector<math_t> norm2_input_vector(0, stream);
+  rmm::device_uvector<math_t> mu_labels_vector(0, stream);
 
   if (fit_intercept) {
-    raft::allocate(mu_input, n_cols);
-    raft::allocate(mu_labels, 1);
+    mu_input_vector = rmm::device_uvector<math_t>(n_cols, stream);
+    mu_labels_vector = rmm::device_uvector<math_t>(1, stream);
+    mu_input = mu_input_vector.data();
+    mu_labels = mu_labels_vector.data();
     if (normalize) {
-      raft::allocate(norm2_input, n_cols);
+      norm2_input_vector = rmm::device_uvector<math_t>(n_cols, stream);
+      norm2_input = norm2_input_vector.data();
```

Comment:
norm2_input will be unassigned if this normalize path isn't taken, right?

Reply:
Yes but this is normal, it is only accessed in pre-processing and post-processing, where the value of `normalize` is checked before acessing this variable.